Update information about xproj migration (#12785)
dotnet/docs: docs/core/migration/index.md

Files that were migrated (*project.json*, *global.json*, *.xproj*, and solution file) are moved to a *Backup* folder. The migrated solution file is upgraded to Visual Studio 2017 or Visual Studio 2019 and you won't be able to open that solution file in Visual Studio 2015 or earlier versions. A file named *UpgradeLog.htm* that contains a migration report is also saved and opened automatically.

> [!IMPORTANT]

> > The above tooling is only available between Visual Studio 2017 (version 15.0) and Visual Studio 2019 (version 16.2). If you are using a version of Visual Studio outside of that range, you should either install a suitable version of Visual Studio or use the command line migration tool described below.

Reviewer: ```suggestion
> In Visual Studio 2019 version 16.3 and later, you cannot load or migrate an *.xproj* file. Additionally, Visual Studio 2015 doesn't provide the ability to migrate an *.xproj* file. If you're using one of these Visual Studio versions, either install a suitable version of Visual Studio, or use the command line migration tool that's described next.
```

A file named *UpgradeLog.htm* is also saved and automatically opened that contains a migration report.

> [!IMPORTANT]

> > The above tooling is only available between Visual Studio 2017 (version 15.0) and Visual Studio 2019 (version 16.3). If you are using a version of Visual Studio outside of that range, you should either install a suitable version of Visual Studio or use the command line migration tool described below.

Reviewer 2: It won't be in 16.3 right? So, this should say 16.2?

Author: Thanks, fixed.

Files that were migrated (*project.json*, *global.json*, and *.xproj*) are moved to a *backup* folder.

> [!NOTE]

> > If you are using Visual Studio Code, the `dotnet migrate` command will not modify Visual Studio Code-specific files such as *tasks.json*. These files need to be changed manually.

Reviewer: ```suggestion
> If you are using Visual Studio Code, the `dotnet migrate` command does not modify Visual Studio Code-specific files such as *tasks.json*. These files need to be changed manually.
```

The MS style guide says to use present tense.

> This is also true if you are using an editor or Integrated Development Environment (IDE) other than Visual Studio.

See [A mapping between project.json and csproj properties](../tools/project-json-to-csproj.md) for a comparison of *project.json* and *.csproj* formats.

If you get an error:

> No executable found matching command dotnet-migrate

Run `dotnet --version` to see which version you are using. [`dotnet migrate`](../tools/dotnet-migrate.md) was introduced in .NET Core SDK 1.0.0 and removed in version 3.0.100.

> You'll get this error if you have a *global.json* file in the current or parent directory and the `sdk` version is set to a version outside this range.

Reviewer: ```suggestion
You'll get this error if you have a *global.json* file in the current or parent directory, and the `sdk` version it specifies is outside this range.
```

Files that were migrated (*project.json*, *global.json* and *.xproj*) will be moved to a *backup* folder.

> [!NOTE]
> If you are using Visual Studio Code, the `dotnet migrate` command will not modify Visual Studio Code-specific files such as *tasks.json*. These files need to be changed manually.
> This is also true if you are using an editor or Integrated Development Environment (IDE) other than Visual Studio.

See [A mapping between project.json and csproj properties](../tools/project-json-to-csproj.md) for a comparison of *project.json* and *.csproj* formats.

If you get an error:

> No executable found matching command dotnet-migrate

> Run `dotnet --version` to see which version you are using. [`dotnet migrate`](../tools/dotnet-migrate.md) is only available between .NET Core CLI RC3 and 3.0.100.

Reviewer 2: It won't be in 3.0.100. That's where we pulled it out.

Author: Fixed. Thanks again.

> No executable found matching command dotnet-migrate

> Run `dotnet --version` to see which version you are using. [`dotnet migrate`](../tools/dotnet-migrate.md) was introduced in .NET Core CLI RC3 and removed in version 3.0.100.

Reviewer 3: RC3? What is RC3? If it's 1.0/1.1 and a release candidate, I would just state 1.0 (or whatever) and omit the preview/rc stuff.

Author: Fair point. @livarcocc can you suggest better text here?

Reviewer 2: It was introduced in .NET Core SDK 1.0.0? I don't think we need to call out exactly the preview version where it first came out.

Author: Thanks, updated.
